packages/playwright/src/common/process.ts

```diff
@@ -92,19 +93,22 @@ process.on('message', async (message: any) => {
   }
 });
 
-async function gracefullyCloseAndExit() {
-  if (closed)
-    return;
-  closed = true;
-  // Force exit after 30 seconds.
-  // eslint-disable-next-line no-restricted-properties
-  setTimeout(() => process.exit(0), 30000);
-  // Meanwhile, try to gracefully shutdown.
-  await processRunner?.gracefullyClose().catch(() => {});
-  if (processName)
-    await stopProfiling(processName).catch(() => {});
-  // eslint-disable-next-line no-restricted-properties
-  process.exit(0);
+async function gracefullyCloseAndExit(forceExit: boolean) {
+  if (forceExit && !forceExitInitiated) {
+    forceExitInitiated = true;
+    // Force exit after 30 seconds.
+    // eslint-disable-next-line no-restricted-properties
+    setTimeout(() => process.exit(0), 30000);
+  }
+  if (!gracefullyCloseCalled) {
+    gracefullyCloseCalled = true;
+    // Meanwhile, try to gracefully shutdown.
+    await processRunner?.gracefullyClose().catch(() => {});
```

Comment:
If a fixture hangs while shutting down, it will keep the process running now. Can we avoid that?